src/window.rs

```diff
@@ -72,48 +73,68 @@ where
     }
 }
 
-impl<T: Copy + Default> Default for LookupTable<T> {
-    fn default() -> LookupTable<T> {
-        LookupTable([T::default(); 8])
+impl<T: Copy + Default> Default for $name<T> {
+    fn default() -> $name<T> {
+        $name([T::default(); $size])
     }
 }
 
-impl<T: Debug> Debug for LookupTable<T> {
+impl<T: Debug> Debug for $name<T> {
     fn fmt(&self, f: &mut ::core::fmt::Formatter) -> ::core::fmt::Result {
-        write!(f, "LookupTable({:?})", self.0)
+        write!(f, "{:?}(", stringify!($name))?;
+
+        for x in self.0.iter() {
+            write!(f, "{:?}", x)?;
+        }
+
+        write!(f, ")")
     }
 }
 
-impl<'a> From<&'a EdwardsPoint> for LookupTable<ProjectiveNielsPoint> {
+impl<'a> From<&'a EdwardsPoint> for $name<ProjectiveNielsPoint> {
     fn from(P: &'a EdwardsPoint) -> Self {
-        let mut points = [P.to_projective_niels(); 8];
-        for j in 0..7 {
+        let mut points = [P.to_projective_niels(); $size];
+        for j in $conv_range {
             points[j + 1] = (P + &points[j]).to_extended().to_projective_niels();
         }
-        LookupTable(points)
+        $name(points)
     }
 }
 
-impl<'a> From<&'a EdwardsPoint> for LookupTable<AffineNielsPoint> {
+impl<'a> From<&'a EdwardsPoint> for $name<AffineNielsPoint> {
     fn from(P: &'a EdwardsPoint) -> Self {
-        let mut points = [P.to_affine_niels(); 8];
+        let mut points = [P.to_affine_niels(); $size];
         // XXX batch inversion would be good if perf mattered here
-        for j in 0..7 {
+        for j in $conv_range {
             points[j + 1] = (P + &points[j]).to_extended().to_affine_niels()
         }
-        LookupTable(points)
+        $name(points)
     }
 }
 
-impl<T> Zeroize for LookupTable<T>
+impl<T> Zeroize for $name<T>
 where
     T: Copy + Default + Zeroize
 {
     fn zeroize(&mut self) {
-        self.0.zeroize();
+        for x in self.0.iter_mut() {
+            x.zeroize();
+        }
     }
 }
 
+}}  // End macro_rules! impl_lookup_table
+
+// The first one has to be named "LookupTable" because it's used as a constructor for consts.
+impl_lookup_table! {Name = LookupTable,         Size =   8, SizeNeg =   -8, SizeRange = 1 ..   9, ConversionRange = 0 ..   7} // radix-16
```

Comment:
Yes, `SizeNeg` is always the negative of `Size`, because the coefficients for the radix-2^n representation are always 0-centered. I remember having some difficulty with getting it to compile without `SizeNeg`, do to the compiler not knowing for certain that `Size` would be a type that implements `Neg`, but I could try doing away with this?

Reply:
Thanks for explaining! I see how it's hard to guarantee to the compiler that `Size` implements `Neg`... in that case it makes sense to keep `SizeNeg`. It's not a big deal, I was mostly curious why it was there.